ydb/docs/en/core/concepts/optimizer.md

```diff
@@ -0,0 +1,81 @@
+
+The cost optimizer consists of three main components:
+
+* Plan enumerator
+* Cost estimation function
+* Statistics module, which is used to estimate statistics for the cost function
+
+### Plan enumerator
+
+The current Cost-Based Optimizer in {{ ydb-short-name }} enumerates all useful join trees, for which the join conditions are defined. We first build a join hypergraph, where the nodes are tables and edges are join conditions. Depending on how the original query was written, the join hypergraph may have quite different topologies, varying from simple chain-like graphs to complex cliques. The resulting topology of the join graph determines how many possible alternite plans need to be considered by the optimizer.
+
+For example, a star is a common topology in analytical queries, where a main fact table is joined to multiple dimension tables:
+
+```sql
+SELECT
+    P.Brand,
+    S.Country AS Countries,
+    SUM(F.Units_Sold)
+
+FROM Fact_Sales F
+INNER JOIN Dim_Date D    ON (F.Date_Id = D.Id)
+INNER JOIN Dim_Store S   ON (F.Store_Id = S.Id)
+INNER JOIN Dim_Product P ON (F.Product_Id = P.Id)
+
+WHERE D.Year = 1997 AND  P.Product_Category = 'tv'
+
+GROUP BY
+    P.Brand,
+    S.Country
+```
+
+In this query graph, all `Dim...` tables are joined to the `Fact_Sales` fact table:
+![Join Graph](_assets/Star-Schema.png)
+
+Common topologies also include chain and clique. A "chain" is a topology where tables are connected to each other sequentially and each table participates in no more than one join. A "clique" is a fully connected graph where each table is connected to another.
+
+In practice, OLAP queries often have a topology that is a combination of a "star" and a "chain", while complex topologies like a "clique" are very rare.
+
+The topology greatly influences the number of alternative plans that the optimizer needs to consider. Therefore, the cost-based optimizer limits the number of joins that are compared by exhaustive search, depending on the topology of the original plan. The capabilities of exact optimization in {{ ydb-short-name }} are listed in the following table:
+
+| Topology | Number of supported joins |
+| -------- | ------------------------- |
+| Chain | 110 |
+| Star | 18 |
+| Clique | 15 |
+
+{{ ydb-short-name }} uses a modification of the [DPHyp](https://www.researchgate.net/publication/47862092_Dynamic_Programming_Strikes_Back) algorithm to search for the best join order. DPHyp is a modern dynamic programming algorithm for query optimization that avoids enumerating unnecessary alternatives and allows you to optimize plans with `JOIN` operators, complex predicates, and even `GROUP BY` and `ORDER BY` operators.
```

Comment:
```suggestion
{{ ydb-short-name }} uses a modification of the [DPHyp](https://www.researchgate.net/publication/47862092_Dynamic_Programming_Strikes_Back) algorithm to search for the optimal join order. DPHyp is a modern dynamic programming algorithm for query optimization that avoids enumerating unnecessary alternatives and enables the optimization of plans involving `JOIN` operators, complex predicates, and even `GROUP BY` and `ORDER BY` operators.
```

Reply:
Here the changes are worse than the original - replacing simpler terms with passive constructs like 'enables the optimization of' or 'involving'